TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -987,6 +989,14 @@ public static void OnNode(Program node)
                     "\"END PROGRAM\" is missing.", MessageCode.Warning);
             }
 
+            if (node.IsMainProgram && compilerOptions.CheckProgramName)
+            {
+                string shortFilename = Path.GetFileNameWithoutExtension(node.CodeElement.TokenSource.SourceName);
+                if (!node.Name.Equals(shortFilename, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
For performance reason, please use StringComparison.OrdinalIgnoreCase

Reply:
done